internal/nodes/avi_model_l4_translator.go

```diff
@@ -606,6 +620,33 @@ func getL4Rule(key string, svc *corev1.Service) (*akov1alpha2.L4Rule, error) {
 	if l4Rule != nil && l4Rule.Status.Status != lib.StatusAccepted {
 		return nil, fmt.Errorf("referred L4Rule %s is invalid", l4Rule.Name)
 	}
+	svcPortsLen := len(svc.Spec.Ports)
+	listenerPortProtoMap := make(map[string]bool)
+	if len(l4Rule.Spec.Services) != 0 {
+		if len(l4Rule.Spec.Services) != svcPortsLen {
+			err := fmt.Errorf("No. of port definitions in %s l4rule listener spec does not match with the port definitons in %s service", l4RuleName, svc.Name)
```

Comment:
these limitation must be added to our documentation.

Reply:
Yes, I will make all the documentation changes by this week.

Reply:
Changed No. to Number